locust_plugins/embedded_resource_manager.py

```diff
@@ -0,0 +1,179 @@
+from locust.contrib.fasthttp import FastHttpUser
+from lxml import html, etree
+import logging
+import re
+
+
+class HttpUserWithResources(HttpUser):
+    """
+    provides embedded resource management for HttpUser
+    """
+
+    abstract = True
+
+    include_resources_by_default = True
+    default_resource_filter = ".*"
+    bundle_resource_stats = True
+    cache_resource_links = True
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        EmbeddedResourceManager(
+            self,
+            self.include_resources_by_default,
+            self.default_resource_filter,
+            self.bundle_resource_stats,
+            self.cache_resource_links,
+        )
+
+
+class FastHttpUserWithResources(FastHttpUser):
+    """
+    provides embedded resource management for FastHttpUser
+    """
+
+    abstract = True
+
+    include_resources_by_default = True
+    default_resource_filter = ".*"
+    bundle_resource_stats = True
+    cache_resource_links = True
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        EmbeddedResourceManager(
+            self,
+            self.include_resources_by_default,
+            self.default_resource_filter,
+            self.bundle_resource_stats,
+            self.cache_resource_links,
+        )
+
+class EmbeddedResourceManager:
+    """
+    provides features for finding and managing resources embedded in html
+    """
+
+    def __init__(
+        self,
+        user,
+        include_resources_by_default,
+        default_resource_filter,
+        bundle_resource_stats,
+        cache_resource_links,
+    ):
+
+        # store resource links for requests
+        self.cache_resource_links = cache_resource_links
+        self.resource_link_cache = {}
+        # bundle all stats into single line for each request (_resources)
+        self.bundle_resource_stats = bundle_resource_stats
+        self.resource_filter_pattern = re.compile(default_resource_filter)
+        self.include_resources = include_resources_by_default
+        # for finding url links in style tags
+        self.url_link_pattern = re.compile(
+            r".*URL\(\s*('|\")(.*)('|\")\s*\).*",
+            re.IGNORECASE | re.MULTILINE | re.DOTALL,
+        )
+
+        # for finding if a link is partial or full
+        self.full_url_pattern = re.compile("^https?://", re.IGNORECASE)
+
+        self.resource_paths = [
+            '//link[@rel="stylesheet"]/@href',
+            '//link[@rel="Stylesheet"]/@href',
+            '//link[@rel="STYLESHEET"]/@href',
+            "//script/@src",
+            "//img/@src",
+            "//source/@src",
+            "//embed/@src",
+            "//body/@background",
+            '//input[@type="image"]/@src',
+            '//input[@type="IMAGE"]/@src',
+            '//input[@type="Image"]/@src',
+            "//object/@data",
+            "//frame/@src",
+            "//iframe/@src",
+        ]
+
+        self.client = user.client
```

Comment:
self.client is never really used except on the next line, right?

Reply:
yep - it'just getting a reference to client so that the wrapper can be used